lib/kafka/connection.rb

```diff
@@ -89,6 +102,12 @@ def request(api_key, request, response_class)
           end
         end
       end
+    rescue Errno::EPIPE, Errno::ECONNRESET, Errno::ETIMEDOUT => e
+      @logger.error "Connection error: #{e}"
+
+      close
```

Comment:
Yes a backoff timer would be needed. Probably best then to leave the decision to the user as you said.